src/main/java/com/aws/greengrass/util/platforms/Exec.java

```diff
@@ -360,31 +293,29 @@ public Exec withErr(Consumer<CharSequence> o) {
         if (shellDecorator != null) {
             decorated = shellDecorator.decorate(decorated);
         }
-        if (userDecorator != null) {
-            decorated = userDecorator.decorate(decorated);
+        if (userOptions != null) {
+            decorated = userOptions.decorate(decorated);
         }
         return decorated;
     }
 
     /**
      * Execute a command.
      *
-     * @returns the process exit code if it is not a background process.
+     * @return the process exit code if it is not a background process.
      * @throws InterruptedException if the command is interrupted while running.
      * @throws IOException if an error occurs while executing.
      */
     @SuppressWarnings("PMD.AvoidRethrowingException")
-    public Optional<Integer> exec() throws InterruptedException, IOException {
+    public Optional<Integer> exec() throws InterruptedException, IOException  {
         // Don't run anything if the current thread is currently interrupted
         if (Thread.currentThread().isInterrupted()) {
             logger.atWarn().kv("command", this).log("Refusing to execute because the active thread is interrupted");
             throw new InterruptedException();
         }
         final String[] command = getCommand();
         logger.atTrace().kv("command", (Supplier<String>) () -> String.join(" ", command)).log();
-        ProcessBuilder pb = new ProcessBuilder();
-        pb.environment().putAll(environment);
-        process = pb.directory(dir).command(command).start();
```

Comment:
I'd keep the environment and directory setting here in the super.

```
proc = new ProcessBuilder(); 
<set env>
<set dir>
 createProcess(proc, command).start()
```
^ that will the subclass to override behavior if it wants, but it makes the defaults sane.

Reply:
Good idea but the Windows implementation won't use `ProcessBuilder` at all...